### PR TITLE
Remove clients of controller-runtime except InjectClient func 🧗‍♂️

### PR DESCRIPTION
**Pull request by houshengbo**
Co-authored-by: Markus Thömmes \<markusthoemmes@me.com\>
## Proposed Changes

* This PR replaces the clients of controller-runtime with either KS client or kubeclient.

**Release Note**

```release-note

```

### CODE REVIEW

File: test/e2e/knativeservingdeployment_test.go [hunk 1]
```diff
@@ -83,35 +83,34 @@ func deploymentRecreation(t *testing.T, clients *test.Clients, names test.Resour
 		t.Fatalf("No deployment under the namespace %q was found",
 			test.ServingOperatorNamespace)
 	}
-	// Delete the deployments one by one to see if they will be recreated.
-	for _, deployment := range dpList.Items {
-		if err := clients.KubeClient.Kube.AppsV1().Deployments(deployment.Namespace).Delete(deployment.Name,
-			&metav1.DeleteOptions{}); err != nil {
-			t.Fatalf("Failed to delete deployment %s/%s: %v", deployment.Namespace, deployment.Name, err)
-		}
+	// Pick up the first deployment for recreation test.
```
Comment (markusthoemmes) [inline test/e2e/knativeservingdeployment_test.go]: Drop this change from this PR please. There's no reason both would change at once (and thus be rolled back at once).

File: pkg/reconciler/knativeserving/knativeserving_controller.go [hunk 2]
```diff
@@ -176,13 +182,12 @@ func (r *ReconcileKnativeServing) initStatus(instance *servingv1alpha1.KnativeSe
...
-	// Account for https://github.com/kubernetes-sigs/controller-runtime/issues/406
-	gvk := instance.GroupVersionKind()
-	defer instance.SetGroupVersionKind(gvk)
+	afterUpdate, err := r.servingClient.ServingV1alpha1().KnativeServings(instance.Namespace).UpdateStatus(instance)
 
-	if err := r.client.Status().Update(context.TODO(), instance); err != nil {
+	if err != nil {
 		return err
 	}
+	afterUpdate.DeepCopyInto(instance)
```
Comment (markusthoemmes) [inline pkg/reconciler/knativeserving/knativeserving_controller.go]: Please leave a TODO to fix this as I did on my commit.

File: pkg/reconciler/knativeserving/knativeserving_controller.go [hunk 3]
```diff
@@ -145,6 +150,7 @@ func (r *ReconcileKnativeServing) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
+	instance.SetGroupVersionKind(servingv1alpha1.SchemeGroupVersion.WithKind("KnativeServing"))
```
Comment (markusthoemmes) [inline pkg/reconciler/knativeserving/knativeserving_controller.go]: Please leave a TODO to have a second look at this, like I did on my commit.

File: pkg/reconciler/knativeserving/knativeserving_controller.go [hunk 4]
```diff
@@ -145,6 +150,8 @@ func (r *ReconcileKnativeServing) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
+	// TODO: We need to fidn a better way to make sure the instance has the updated info.
```
Comment (mattmoor-sockpuppet) [inline pkg/reconciler/knativeserving/knativeserving_controller.go]:
```suggestion
	// TODO: We need to find a better way to make sure the instance has the updated info.
```

**Comment by houshengbo** [comment]
@markusthoemmes only the function  InjectClient is left.

**Pull request merged**